### PR TITLE
[mlir] IntegerRangeAnalysis: don't loop over splat attr 

### PR DESCRIPTION
**Pull request by IanWood1**
If the `DenseIntElementsAttr` is a splat value, there is no need to loop over the entire attr. Instead, just update with the splat value.

### CODE REVIEW

**Review by Hardcode84 — COMMENTED** [review]
Can you add some test? LGTM otherwise.

File: mlir/lib/Dialect/Arith/IR/InferIntRangeInterfaceImpls.cpp [hunk 1]
```diff
@@ -42,13 +42,19 @@ void arith::ConstantOp::inferResultRanges(ArrayRef<ConstantIntRanges> argRanges,
   }
   if (auto arrayCstAttr =
           llvm::dyn_cast_or_null<DenseIntElementsAttr>(getValue())) {
+    assert(arrayCstAttr.size() && "Zero-sized vectors are not allowed");
+    if (arrayCstAttr.isSplat()) {
+      setResultRange(getResult(), ConstantIntRanges::constant(
```
Comment (kuhar) [inline mlir/lib/Dialect/Arith/IR/InferIntRangeInterfaceImpls.cpp:47]: you can and an `if` before this one and check using `dyn_cast_or_null<SplatElementsAttr>(...)`

Comment (IanWood1) [reply mlir/lib/Dialect/Arith/IR/InferIntRangeInterfaceImpls.cpp]: Nice suggestion, done!

**Comment by IanWood1** [comment]
> Can you add some test? LGTM otherwise.

[f54cdc5](https://github.com/llvm/llvm-project/commit/f54cdc5d6ee5532da117f2489c105148c94dcb39) added tests to check the for the correct handling of splat values. This change should really only effect the performance of very large splat constants. So, I'm not sure how this should be tested.

**Review by kuhar — APPROVED** [review]

**Comment by Hardcode84** [comment]
> > Can you add some test? LGTM otherwise.
> 
> [f54cdc5](https://github.com/llvm/llvm-project/commit/f54cdc5d6ee5532da117f2489c105148c94dcb39) added tests to check the for the correct handling of splat values and change should really only affect the performance of very large splat constants. So, I'm not sure how this should be tested.

Oh, yeah, I forgot we already had functional test for splats ). LGTM.

**Review by Hardcode84 — APPROVED** [review]

**Review by krzysz00 — APPROVED** [review]
Approved, since there's already a `dense<3>` in the vector test suite

**Comment by joker-eph** [comment]
>  change should really only affect the performance of very large splat constants. So, I'm not sure how this should be tested.

In this case we tag the PR with [NFC] (no functional change) to convey this.

**Pull request merged**